src/rgw/rgw_common.h

```diff
@@ -2669,4 +2669,16 @@ static inline string rgw_bl_str(ceph::buffer::list& raw)
   return s;
 }
 
+template <typename T>
+int decode_bl(bufferlist& bl, T& t)
+{
+  auto iter = bl.cbegin();
+  try {
+    t.decode(iter);
```

Comment:
`decode(t, iter);` seems like the canonical way to do this part, and doesn't require T to have member functions

Reply:
ack